### DISCUSSION TITLE
Document environment variable usage with custom server

### DISCUSSION THREAD

**Discussion by bradencanderson** [vercel/next.js#23963] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

It's not immediately obvious that env vars declared in `.env` are only available in nextjs-bundled code, and not in the custom server code.

## Describe the solution you'd like

I think a short explanation of the current behavior and instructions to call `dotenv` manually might be helpful. Perhaps by modifying the note on babel/webpack:

> `server.js` doesn't go through babel or webpack, and it doesn't inherit [environment variables](https://nextjs.org/docs/basic-features/environment-variables) from your next.js app.

## Describe alternatives you've considered

Perhaps a revisiting of how custom servers work.

## Additional context

A solution I'm using is to add this to the top of my `server.js`:
```
const path = require("path");
const dotenv = require("dotenv");
const dotenvExpand = require("dotenv-expand");
// Set env vars defined in .env.* files
const env = dotenvExpand(
  dotenv.config({
    path: path.resolve(
      process.cwd(),
      process.env.NODE_ENV === "production"
        ? ".env.production"
        : ".env.development"
    ),
  })
);
```

**Comment by dryleaf** [comment]
Any solutions to this issue? @bradencanderson Did you find a better solution to this?

**Comment by dryleaf** [comment]
@timneutkens What are thoughts on this issue?

**Comment by dryleaf** [comment]
Anyways, you do not need to use dotenv or dotenvExpand, you can simply pass the environment variables by incorporating `env $(cat .env.development .env | grep -v \"#\" | xargs)` or `env $(cat .env.production .env | grep -v \"#\" | xargs)` as follows:
```
"scripts": {
    "dev": "env $(cat .env.development .env | grep -v \"#\" | xargs) node server.js",
    "build": "next build",
    "start": "cross-env NODE_ENV=production env $(cat .env.production .env | grep -v \"#\" | xargs) node server.js"
  },
```
